### ISSUE TITLE
ImportError: no module named pade.core.agent

### ISSUE THREAD

**Issue by Relex12** [grei-ufc/pade#68]
Hi everyone, here is a description of the trouble I encounter:

I try to run a simple agent as described in the [pade.readthedocs.io main page](https://pade.readthedocs.io/en/latest/index.html#pade-e-simples):

```
adrian@DESKTOP-SO75EQB:$ pade start-runtime --port 20000 hello-world-pade.py
:0: UserWarning: You do not have a working installation of the service_identity module: 'cannot import name 'verify_ip_address''.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
please enter a username [pade_user]:
Password [12345]:
[...] Creating Pade tables in selected data base.
```
Rigth after, the screen turns into this:
```
        This is
         ____   _    ____  _____
        |  _ \ / \  |  _ \| ____|
        | |_) / _ \ | | | |  _|
        |  __/ ___ \| |_| | |___
        |_| /_/   \_\____/|_____|

        Python Agent DEvelopment framework


        PADE is a free software under development by:

        - Electric Smart Grid Group - GREI
        Federal University of Ceara - UFC - Brazil

        - Laboratory of Applied Artificial Intelligence - LAAI
        Federal University of Para - UFPA

        https://github.com/grei-ufc/pade
 * Serving Flask app "pade.web.flask_server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
Traceback (most recent call last):
  File "/home/adrian/.local/lib/python3.6/site-packages/pade/core/new_ams.py", line 25, in <module>
    from pade.core.agent import Agent_
ImportError: No module named pade.core.agent
Traceback (most recent call last):
  File "/home/adrian/.local/lib/python3.6/site-packages/pade/core/sniffer.py", line 26, in <module>
    from pade.core.agent import Agent
ImportError: No module named pade.core.agent
Traceback (most recent call last):
  File "hello-world-pade.py", line 7, in <module>
    from pade.core.agent import Agent
ImportError: No module named pade.core.agent
``` 

After this, nothing happens.

^C gives "Stoping PADE..." and then a AttributeError: 'FlaskServerProcess' object has no attribute 'kill'

^Z brings the terminal to normal

Here is the hello-world-pade.py:

```
#!coding=utf-8

from pade.core.agent import Agent
from pade.misc.utility import display_message, start_loop
from pade.acl.aid import AID
from sys import argv

class AgenteHelloWorld(Agent):
    def __init__(self, aid):
        super(AgenteHelloWorld, self).__init__(aid=aid)
        display_message(self.aid.localname, 'Hello World!')


if __name__ == '__main__':

    agents_per_process = 3
    c = 0
    agents = list()
    for i in range(agents_per_process):
        port = int(argv[1]) + c
        agent_name = 'agent_hello_{}@localhost:{}'.format(port, port)
        agente_hello = AgenteHelloWorld(AID(name=agent_name))
        agents.append(agente_hello)
        c += 1000

    start_loop(agents)
```

I tried to do the imports in the Python interpreter:

```
adrian@DESKTOP-SO75EQB:$ python3
Python 3.6.9 (default, Jan 26 2021, 15:33:00)
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pade.core.agent import Agent
:0: UserWarning: You do not have a working installation of the service_identity module: 'cannot import name 'verify_ip_address''.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
>>> from pade.core.agent import Agent
>>> from pade.misc.utility import display_message, start_loop
>>> from pade.acl.aid import AID
>>>
```

Note that the UserWarning occurs only for the first pade import, no matter wich one it is.

I tried to install service_identity, but it seems to be already installed:

```
adrian@DESKTOP-SO75EQB:$ python3 -m pip install service_identity
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: service_identity in /usr/lib/python3/dist-packages (16.0.0)
adrian@DESKTOP-SO75EQB:$
```

Note that I don't know why normal site-packages is not writeable (this might be a part of the problem).

I installed PADE from PIP, and it went pretty well.

For information, I am working on Ubuntu 18.04 LTS with a WSL.

As I am new to PADE, it might be a simple error, but I didn't find anything about this.

If you need more information, please ask me and give me the commands I must run.

Thank you !

**Comment by Relex12** [comment]
I found what was the problem.

As PADE is launched from the CLI, it very likely executes a `python` command, wich refers to Python2 if installed, as it is on Ubuntu 18.04.

I am not using Python2 anywhere, so I did `sudo ln -sf /usr/bin/python3 /usr/bin/python` to make `python` always point to Python3.

I am leaving the issue open cause I don't think that is a normal behaviour and I think it should be fixed.

If one of the developers disagrees, and thinks it is a feature and not a bug, he can close the issue.




About the this warning:
```
:0: UserWarning: You do not have a working installation of the service_identity module: 'cannot import name 'verify_ip_address''.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
```

As discussed [here](https://stackoverflow.com/questions/47681437/python-scrapy-service-identityopentype-not-working-and-cannot-install/54337295), I ran `python -m pip install service_identity --force --upgrade`.

It gave me some errors, but it still fixed the problem.
